cdmtaskservice/dockerimages.py

```diff
@@ -2,29 +2,151 @@
 Contains code for querying container information without access to a docker server
 """
 
+# Trying to find a simple way to handle this was surprisingly frustrating. I spent a half
+# day or so looking at alternatives and the crane CLI seems like the least bad option. I tried
+# a number of things, most of which I don't remember, including http requests (had to be tailored
+# to each repository somewhat) and https://github.com/davedoesdev/dxf (couldn't get anonymous
+# access to work). If there's a python library that can take crane's place that'd be great but
+# I didn't find it.
+
+import asyncio
+import logging
+import os
 from pathlib import Path
+from typing import Self
+
+from fqdn import FQDN
+
 
 class DockerImageInfo:
     """
     Provides information about a remote docker image
     """
     
-    def __init__(self, crane_absolute_path: Path):
+    @classmethod
+    async def create(cls, crane_absolute_path: Path) -> Self:
         """
-        Initialize the image info provider.
+        Create the image info provider.
         
         crane_absolute_path - the absolute path to the crane executable.
         """
-        # Require an absolute path to avoid various malicious attacks
-        # TODO check is absolute path
-        # TODO run crane sync and check version, store path
+        dii = DockerImageInfo(crane_absolute_path)
+        retcode, _, stde = await dii._run_crane("version")
```

Comment:
Any executable (e.g., `git`, `docker`, etc.) that supports a valid `version` command can pass this validation.

Reply:
Yeah, this is more about making sure it can run and is there

Reply:
👍 I wasn't able to resolve comments. Maybe I am not added to the repo yet.

Reply:
fixed

Reply:
👍